charts/argo-cd/templates/argocd-application-controller/deployment.yaml

```diff
@@ -0,0 +1,108 @@
+{{- if .Values.controller.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
```

Comment:
This (and other namespaced resources) should specify

```yaml
namespace: {{ Release.Namespace }}
```

in order to support installing with `helm template`.

Reply:
I would disagree for a couple reasons.
First, resources that require a namespace in their spec (roles / cluster roles) have it defined as the {{ Release.Namespace }} however it is not standard for other resources to have this defined.  To define standard I am using the common practices of the helm/stable repository specify charts for graduated or incubating cncf.io projects.  
Second, ArgoCD itself doesn't require items to have namespaces defined during deployment.  It will template items and put them in the namespace defined by the application spec.

Reply:
The usage of `{{ Release.Namespace}}` in templates is not consistent, I'll give you that, but I'd argue that it should be a standard for most charts, especially those that use it elsewhere in templates as in the case of roles, etc.

I can't think of a valid use-case where one would do something like `helm template --namespace foo | kubectl apply -f -` and not expect the templates to be installed in the provided namespace, but that's how it is today.  In order to install things correctly, you actually need to do something along the lines of `helm template --namespace foo | kubectl apply -n foo -f -` which seems redundant.

You're completely correct that ArgoCD itself will place resources in the correct namespace, however this chart is for installing ArgoCD, so you have to have some way to bootstrap the installation; there has to be a first install of ArgoCD somewhere.  Adding the namespace explicitly reduces the difference between `helm install argocd --namespace argocd` and `helm template argocd --namespace argocd | kubectl apply -f -`.

This admittedly is far less of an issue in Helm 3 where the lack of Tiller makes it more reasonable to skip using `helm template`, but something to consider.

Reply:
I agree that it can be re-evaluated in helm 3, though for the sake of keeping things consistent with other major projects with helm charts there isn't a compelling reason to stray from the standard. 
I would be of a different opinion if ArgoCD needed to be installed in multiple namespaces but I don't think that is the case at the moment.
When/if that does happen I would strongly suggest not using `{{ Release.Namspace }}` but using a templated `namespaceOverride` value as in the Jenkins chart https://github.com/helm/charts/pull/15202 because it adds a good amount of flexibility for those who only use help as a templating tool.